mesonbuild/build.py

```diff
@@ -873,6 +873,13 @@ def process_compilers_late(self) -> None:
         # did user override clink_langs for this target?
         link_langs = [self.link_language] if self.link_language else clink_langs
 
+        if not any(self.all_compilers.get(lang) for lang in link_langs):
```

Comment:
I think `lang in self.all_compilers` is more idiomatic.

But, I wonder if something like `if not set(link_langs).intersection(self.all_compilers)` would be better? I'm not sure on that one.